File: pointer_and_dynamic_array/bag/php/driver.php
<?php
require_once __DIR__.'/bag.php';

// -------------------------- script --------------------------
$numbers;
$array_size = 0;
$mean_value = 0;

echo "This program will compute the average of some numbers. The\n";
echo "numbers will be stored in an array of doubles that I allocate.\n";
echo "How many doubles should I allocate?\n";
do {
    echo "Please type a positive integer answer:\n";
    fscanf(STDIN, "%d\n", $array_size);
} while(!is_numeric($array_size));
$numbers = new bag($array_size);
$numbers->fill();
$numbers->print_final_sequence();

File: pointer_and_dynamic_array/bag/c++/driver.cpp
#include<iostream>
#include<cstdlib>
#include<cassert>
#include "bag.h"

using namespace std;
using namespace bag_h;


int main(int argc, char* argv[]){
    size_t array_size;
    double mean_value;

    cout << "This program will compute the average of some numbers. The\n";
    cout << "numbers will be stored in an array of doubles that I allocate.\n";
    cout << "How many doubles should I allocate?" << endl;
    cout << "Please type a positive integer answer: ";
    cin >> array_size;
    bag* numbers = new bag(array_size);

    numbers->fill(array_size);

    numbers->print_final_sequence();
    return EXIT_SUCCESS;
}

